src/secontext.c

```diff
@@ -137,10 +139,11 @@ selinux_getfdcon(pid_t pid, int fd, char **secontext, char **expected)
 	 * may be reused by a different file with different context.
 	 */
 	strace_stat_t st;
-	if (stat_file(linkpath, &st))
+	if (stat_file(buf, &st) < 0)
```

Comment:
Sorry, I don't follow this particular change.  Isn't stat(linkpath) more reliable in this case than stat(buf)?

Reply:
We want here the context of the target pointed by the symlink `/proc/<pid>/fd/<fd>`, not the symlink itself, hence the stat on `buf`.

Reply:
As stat_file() follows symlinks, there shouldn't bean issue, unless the file is deleted, in that case stat(buf) wouldn't be able to access it.